### DISCUSSION TITLE
Error when execute preproc_batch_tops

### DISCUSSION THREAD

**Discussion by Abdallahbalushi** [gmtsar/gmtsar#855] [Q&A]
When I am trying to do this step I got an error that the orbits xml files could not be opened. 
![image](https://github.com/gmtsar/gmtsar/assets/150544398/bad4d200-bf96-4601-b46e-41aa10290ea8)
![image](https://github.com/gmtsar/gmtsar/assets/150544398/3e683fdd-59cc-4e63-950b-c1f7644468d8)
Then, I checked my data.in file and it seems not correct, so I corrected each image with its orbit manually, however it still says that no orbit coverage in the selected area. 
![image](https://github.com/gmtsar/gmtsar/assets/150544398/a5e164b5-300a-4fa1-af23-8848b9424c82)
Any ideas please?

**Comment by Xiaohua-Eric-Xu** [comment]
Could it be due to the orbit format change from ESA? https://github.com/gmtsar/gmtsar/discussions/629

Which version of GMTSAR are you using?

**Reply by Abdallahbalushi** [reply]
Hi Eric, 

I am using version 6.3.0

**Reply by Xiaohua-Eric-Xu** [reply]
Then could this be a permission issue? i.e., are you able to access the .EOF file?

**Reply by Abdallahbalushi** [reply]
Hey Eric, 
I do not think it is a permission issue as I am able to access those .EOF file via the notepad.
What possibly do you think is the issue here?

**Reply by Abdallahbalushi** [reply]
@Xiaohua-Eric-Xu, 
So, until now I faced multiple issues. 
Firstly, downloading precise orbit files through a command >> so I downloaded them manually.
Then, when tring to produce a data.in file using prep_data_linux.csh, it mix the images with the wrong precise orbits that have been downloaded manually >> so I tend to also produce a data.in file manually with the correct corresponding orbit file. 
After that, and **most importantly**, this issue now with preproc_batch_tops.csh for mode 1 & 2. 
I am still facing the same error with no orbit coverage in the selected area.

Any ideas??

**Reply by Xiaohua-Eric-Xu** [reply]
The following error 
```
Couldn't open xml file ...
```
Comes from this line in the c code ext_orb_s1a.c
```
 48     if ((XML_FILE = fopen(argv[2], "r")) == NULL)
 49         die("Couldn't open xml file: \n", argv[2]);
```
Line 48 is only opening the file with `fopen` and got denied so it popped out the line below. The probable cause for denying access to a file is likely permission issue. Please make sure you can open and read the .EOF files located in your raw directory.
